tests/rest/client/test_rooms.py

```diff
@@ -832,6 +832,25 @@ async def user_may_join_room(
 
         self.assertEqual(join_mock.call_count, 0)
 
+        async def user_may_join_room_tuple(
+            mxid: str,
+            room_id: str,
+            is_invite: bool,
+        ) -> Tuple[Codes, dict]:
+            return (Codes.CONSENT_NOT_GIVEN, {})
+
+        join_mock = Mock(side_effect=user_may_join_room_codes)
+        self.hs.get_spam_checker()._user_may_join_room_callbacks = [join_mock]
```

Comment:
You could also replace this with:

```suggestion
        join_mock.side_effect = user_may_join_room_codes
```